src/geometry.F90

```diff
@@ -956,9 +958,13 @@ subroutine handle_lost_particle(p, message)
 !$omp atomic
     n_lost_particles = n_lost_particles + 1
 
+    ! Count the total number of simulated particles
+    tot_n_particles = n_batches * gen_per_batch * n_particles
```

Comment:
Right now the number of lost particles is not reduced across multiple processes, so that should probably be reflected in the total number of particles.